docs/BestPractices.md

```diff
@@ -21,6 +21,13 @@ This is because the input stream is a direct stream of data from the HTTP connec
 reused until all data from the stream has been read and the stream is closed. If these rules are not followed, the client can 
 run out of resources by allocating too many open, but unused, HTTP connections.
 
+### Close input streams of responseBody() from Custom Client based on SdkHttpClient
```

Comment:
Added JavaDoc to HttpExecuteResponse. I believe it would be beneficial to include this information in [BestPractices.md](https://github.com/aws/aws-sdk-java-v2/pull/4624/files/89544a01399a44d0438dcbbdbf0a324662440f2c#diff-206cb65fb778070642816334d5f181cc41bc5e8bd46ac9b8fe1e91373dd652ce) as well. As of now, I don't foresee any adverse effects from documenting this in BestPractices.md

Reply:
Ideally, we should not let them diverge, but it's hard to make sure we always update both places. From my experience, dev guide is more discoverable than GH and that's why I suggested linking to the dev guide in this page so that we have a single source of truth. The majority of SDK customers don't use this API directly, so I'm not sure if it's worth adding it to best practices page, but I suppose there's not much harm doing that; it's just that we need to keep in mind best practices should be concise, and if it's too long, people may not read it.